tests/test_black.py

```diff
@@ -910,6 +910,18 @@ def test_invalid_include_exclude(self) -> None:
             result = CliRunner().invoke(black.main, ["-", option, "**()(!!*)"])
             self.assertEqual(result.exit_code, 2)
 
+    def test_preserves_line_endings(self) -> None:
+        with TemporaryDirectory() as workspace:
+            test_file = Path(workspace) / "test.py"
+            for nl in ["\n", "\r\n"]:
+                contents = nl.join(["def f(  ):", "    pass"])
+                test_file.write_bytes(contents.encode())
+                ff(test_file, write_back=black.WriteBack.YES)
+                updated_contents: bytes = test_file.read_bytes()
+                self.assertIn(nl.encode(), updated_contents)  # type: ignore
```

Comment:
mypy complains about this line like so:
```
Cannot infer type argument 1 of "assertIn" of "TestCase"
```
I think it's confused because both arguments are `bytes`. Seems like a bug in mypy to me

Reply:
I think the typeshed stub is wrong. It has:
```
    def assertIn(self, first: _T, second: Iterable[_T],
                 msg: Any = ...) -> None: ...
    def assertNotIn(self, first: _T, second: Iterable[_T],
                    msg: Any = ...) -> None: ...
```
This should actually be `Container`, not `Iterable`.

Reply:
https://github.com/python/typeshed/pull/2186 should fix it.